core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -141,4 +133,18 @@ private static String stripTrailingSlash(String path) {
     }
     return result;
   }
+
+  public static String dataLocation(Map<String, String> properties, String tableLocation) {
```

Comment:
Nit: Does this need to be `public` or can it be `private`? If it needs to be visible for testing or something, that's ok.

Reply:
Visible for testing should use protected or package-private.

Reply:
I make it public because it is used in IcebergSourceBenchmark.java. Maybe we can change `WRITE_FOLDER_STORAGE_LOCATION` to the `WRITE_DATA_PATH` in IcebergSourceBenchmark.java?